includes/discovery/entity-physical/entity-physical.inc.php

```diff
@@ -25,6 +65,28 @@
     echo ' hwEntityBomEnDesc';
     $entity_array = snmpwalk_cache_oid($device, 'hwEntityBomEnDesc', $entity_array, 'ENTITY-MIB:HUAWEI-ENTITY-EXTENT-MIB');
 }
+if ($device['os'] == 'saf-cfml4') {
+    $entity_array = array();
+    $device_array = array();
+    echo ' saf-cfml4Anatomy';
+    $oid = '.1.3.6.1.4.1.7571.100.1.1.2.22';
+    $row = 0;
+    $device_array = snmpwalk_cache_oid($device, $oid, $entity_array, 'SAF-MPMUX-MIB');
+    // Descr, VendorType, ContainedIn, Class, ParentRelPos,
+    //  Name, HardwareRev, FirmwareRev, SoftwareRev, SerialNum, MfgName, ModelName, Alias, Alias, AssetID, IsFRU
+    $entity_array[++$row] = return_entity_array('CFM L4', 'CFM L4', '0', 'chassis', '-1', 'Chassis', '', '', '', $device_array[0]['serialNumber'], 'SAF', 'CFM L4', '', '', 'true');
```

Comment:
1. You  don't need to use the $row variable `$entity_array[] = 'something';` just appends it at the next index.
2. foreach would be a lot cleaner instead of for (what if the device has  a different number of responses than you expect.
3. You could probably loop through this in a way that you wouldn't need the function and still have it clean looking.